archaius-core/src/main/java/com/netflix/archaius/config/InterpolatingConfig.java

```diff
@@ -0,0 +1,347 @@
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public BigDecimal getBigDecimal(String key) {
+        String value = interpolate(key);
+        if (value == null) 
+            return notFound(key);
+        try {
+            return new BigDecimal(value);
+        }
+        catch (NumberFormatException e) {
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public BigDecimal getBigDecimal(String key, BigDecimal defaultValue) {
+        String value = interpolate(key);
+        if (value == null) 
+            return notFound(key, defaultValue);
+        try {
+            return new BigDecimal(value);
+        }
+        catch (NumberFormatException e) {
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public Float getFloat(String key) {
+        String value = interpolate(key);
+        if (value == null) 
+            return notFound(key);
+        try {
+            return Float.parseFloat(value);
+        }
+        catch (NumberFormatException e) {
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public Float getFloat(String key, Float defaultValue) {
+        String value = interpolate(key);
+        if (value == null) 
+            return notFound(key, defaultValue);
+        try {
+            return Float.parseFloat(value);
+        }
+        catch (NumberFormatException e) {
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public Byte getByte(String key) {
+        String value = interpolate(key);
+        if (value == null) 
+            return notFound(key);
+        try {
+            return Byte.parseByte(value);
+        }
+        catch (NumberFormatException e) {
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public Byte getByte(String key, Byte defaultValue) {
+        String value = interpolate(key);
+        if (value == null) 
+            return notFound(key, defaultValue);
+        try {
+            return Byte.parseByte(value);
+        }
+        catch (NumberFormatException e) {
+            return parseError(key, value, e);
+        }
+    }
+
+    @Override
+    public List getList(String key) {
+        String value = interpolate(key);
+        if (value == null) {
+            return notFound(key);
+        }
+        String[] parts = value.split(",");
+        return Arrays.asList(parts);
+    }
+
+    @Override
+    public List getList(String key, List defaultValue) {
+        String value = interpolate(key);
+        if (value == null) {
+            return notFound(key, defaultValue);
+        }
+        String[] parts = value.split(",");
+        return Arrays.asList(parts);
+    }
+
+    @Override
+    public <T> T get(Class<T> type, String key) {
+        return get(type, key, null);
+    }
+
+    @Override
+    public <T> T get(Class<T> type, String key, T defaultValue) {
```

Comment:
Is there any reason the other getXXX methods cannot delegate to the one for the generic class? For example:

```java
@Override
public Short getShort(String key, Short defaultValue) {
    return get(Short.class, key, defaultValue);
}
```